apm-agent-core/src/main/java/co/elastic/apm/agent/impl/context/Db.java

```diff
@@ -40,7 +40,7 @@ public class Db implements Recyclable {
         new Allocator<CharBuffer>() {
             @Override
             public CharBuffer createInstance() {
-                return CharBuffer.allocate(DslJsonSerializer.MAX_LONG_STRING_VALUE_LENGTH);
+                return CharBuffer.allocate(SerializationConstants.getMaxLongStringValueLength());
```

Comment:
This is not 100% spec conformant because some unicode characters might require two UTF-16 chars to be represented ([wiki link](https://en.wikipedia.org/wiki/UTF-16#Code_points_from_U+010000_to_U+10FFFF)), but this is such an edge case that I think this is acceptable.

Reply:
Yeah, we can definitely live with that. 
Since this is a configuration to limit field size for transport and storage, it even makes more sense this way, assuming that it will enforce the maximum size we want in all encoded forms.